File: python3/sdk/asr_sentence_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.gettoken import get_token
from ais_sdk.utils import encode_to_base64
from ais_sdk.asr_sentence import asr_sentence
from ais_sdk.asr_sentence import asr_sentence_aksk

if __name__ == '__main__':
    #
    # access asr, asr_sentence,post data by token
    #
    user_name = '******'
    password = '******'
    account_name = '******'  # the same as user_name in commonly use

    demo_data_url = 'https://ais-sample-data.obs.myhwclouds.com/asr-sentence.wav'
    token = get_token(user_name, password, account_name)

    # call interface use the url
    result = asr_sentence(token, '', demo_data_url, 'wav', '16k')
    print (result)

    # call interface use the file
    result = asr_sentence(token, encode_to_base64('data/asr-sentence.wav'), '', 'wav', '16k')
    print (result)

    #
    # access asr, asr_sentence,post data by ak,sk
    #
    app_key = "*************"
    app_secret = "************"

    # call interface use the url
    result = asr_sentence_aksk(app_key, app_secret,'', demo_data_url,'wav', '16k')
    print (result)

    # call interface use the file
    result = asr_sentence_aksk(app_key, app_secret, encode_to_base64('data/asr-sentence.wav'), '', 'wav', '16k')
    print (result)

File: php/asr_sentence_demo.php
<?php
/**
 * 语音识别服务的使用示例
 */
require "./ais_sdk/gettoken.php";
require "./ais_sdk/asr_sentence.php";
require "./ais_sdk/utils.php";

/**
 * token 方式请求
 */
$username = "********";      // 配置用户名
$password = "********";      // 密码
$domainName = "*********";   // 配置用户名
$regionName = "********";    // 配置区域信息

$filepath = "./data/asr-sentence.wav";
$data = fileToBase64($filepath);

$data_url = "https://ais-sample-data.obs.myhwclouds.com/asr-sentence.wav";

$token = gettoken($username, $password, $domainName, $regionName);

// base64 方式请求
$result = asr_sentence($token, $data, "", "wav", "16k");
echo $result;
$resultobj = json_decode($result, true);
echo $resultobj["result"]["words"];

// obs的url方式请求
$result = asr_sentence($token, "", $data_url, "wav", "16k");
echo $result;
$resultobj = json_decode($result, true);
echo $resultobj["result"]["words"];
/**
 * ak,sk 方式请求
 */

$app_key = "*************";
$app_secret = "*************";

// base64 方式请求
$result = asr_sentence_aksk($app_key, $app_secret, $data, "", "wav", "16k");
echo $result;
$resultobj = json_decode($result, true);
echo $resultobj["result"]["words"];

// obs的url方式请求
$result = asr_sentence_aksk($app_key, $app_secret, "", $data_url, "wav", "16k");
echo $result;
$resultobj = json_decode($result, true);
echo $resultobj["result"]["words"];

